### PR TITLE
feat: [CDS-93909]: Add docs for GitOps Agent mTLS

### PR DESCRIPTION
**Pull request by mteodor**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Adding instructions how to set mTLS for GitOps Agent
* Jira/GitHub Issue numbers (if any): https://harness.atlassian.net/browse/CDS-93470
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by ashinsabu3** [comment]
Are we keeping this just in the quickstart section?
https://developer.harness.io/docs/continuous-delivery/gitops/use-gitops/install-a-harness-git-ops-agent/
We can have this here or have a separate section named something like "Configuring mTLS with your gitops agent" or "Harness Gitops agent with mTLS" similar to this https://developer.harness.io/docs/continuous-delivery/gitops/use-gitops/harness-git-ops-agent-with-self-signed-certificates

File: docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md [hunk 1]
```diff
@@ -195,6 +195,32 @@ In your cloud platform Kubernetes cluster you can see the agent workload:
 
 Now that you have the Harness GitOps Agent installed, running, and registered, you can configure the remaining components.
 
+## Installing mTLS GitOps Agent
+
+For installing GitOps Agent to run in mTLS mode we will use GitOps agent helm charts, so we need to download `overrides.ymal` file from the Harness UI.
```
Comment (mankrit-singh) [inline docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md:200]: could you also add that this is behind Feature Flag?

Comment (roshnisarangadharan) [inline docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md:200]:
```suggestion
For installing GitOps Agent to run in mTLS mode, Harness uses GitOps Agent Helm charts. For this, you must download the `overrides.yaml` file from the Harness UI.
```

File: docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md [hunk 2]
```diff
@@ -195,6 +195,32 @@ In your cloud platform Kubernetes cluster you can see the agent workload:
...
+If you want to install a Harness GitOps Agent with mTLS enabled, first follow steps for configuring mTLS for Harness Delegate as described in [Delegate mTLS support](/docs/platform/Delegates/secure-delegates/delegate-mtls-support).
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md:201]:
```suggestion
To install Harness GitOps Agent with mTLS enabled, first follow steps for configuring mTLS for Harness Delegate as described in [Delegate mTLS support](/docs/platform/Delegates/secure-delegates/delegate-mtls-support).
```

File: docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md [hunk 3]
```diff
@@ -195,6 +195,32 @@ In your cloud platform Kubernetes cluster you can see the agent workload:
...
+After following steps in [Delegate mTLS support](/docs/platform/Delegates/secure-delegates/delegate-mtls-support) you should have the following:
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md:202]:
```suggestion
After following steps in [Delegate mTLS support](/docs/platform/Delegates/secure-delegates/delegate-mtls-support), make sure that you have met the following requirements:
```

File: docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md [hunk 4]
```diff
@@ -195,6 +195,32 @@ In your cloud platform Kubernetes cluster you can see the agent workload:
...
+ - CA certificate. 
+ - Client certificate and key created using the previously created CA certificate.
+ - mTLS enabled for your account and host to be used for GitOps Agent connection <YOUR_FQDN>.agent.harness.io
+
+When above requirements are met you can proceed with installation.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md:207]:
```suggestion

```

**Review by roshnisarangadharan — DISMISSED** [review]

**Review by roshnisarangadharan — DISMISSED** [review]

File: docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md [hunk 5]
```diff
@@ -195,6 +195,35 @@ In your cloud platform Kubernetes cluster you can see the agent workload:
 
 Now that you have the Harness GitOps Agent installed, running, and registered, you can configure the remaining components.
 
+## Installing mTLS GitOps Agent
+
+For installing GitOps Agent to run in mTLS mode, Harness uses GitOps Agent Helm charts. For this, you must download the `overrides.yaml` file from the Harness UI.
+
+To install Harness GitOps Agent with mTLS enabled, make sure that you've met the following conditions:
+- Download the `overrides.yaml` file from the Harness UI.
+- Configure mTLS for Harness Delegate[Delegate mTLS support](/docs/platform/Delegates/secure-delegates/delegate-mtls-support)
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/gitops/get-started/harness-cd-git-ops-quickstart.md:204]:
```suggestion
- Configure mTLS for Harness Delegate[Delegate mTLS support](/docs/platform/delegates/secure-delegates/delegate-mtls-support/)
```

**Review by roshnisarangadharan — APPROVED** [review]

**Pull request merged**